featureFlag/dashboard/index.js

```diff
@@ -0,0 +1,152 @@
+  }
+
+  return feature;
+}
+
+const changeTheFeatureToggle = (id) => {
+  const featureBox = document.getElementById(id);
+
+  if (featureBox.querySelector(".ckBoxToggle").classList.contains("checked")) {
+
+    featureBox.querySelector(".featureTitle").classList.add("inactive");
+    featureBox.querySelector(".createdDate").classList.add("inactive");
+    const featureCkbox = featureBox.querySelector(".featureToggle");
+    featureCkbox.setAttribute("checked", "false");
+    const ckboxOverlay = featureBox.querySelector(".primaryInfo2 > span");
+    ckboxOverlay.classList.remove("checked");
+    ckboxOverlay.classList.add("unchecked");
+    featureBox.querySelector(".isActive").innerHTML = "off";
+    featureBox.querySelector(".isActive").classList.add("inactive");
+    featureBox.querySelector(".launchedDate").classList.add("inactive");
+    featureBox.querySelector(".ownerName").classList.add("inactive");
+    featureBox.querySelector(".ownerImg").classList.add("imgInactive");
+
+  } else {
+
+    featureBox.querySelector(".featureTitle").classList.remove("inactive");
+    featureBox.querySelector(".createdDate").classList.remove("inactive");
+    const featureCkbox = featureBox.querySelector(".featureToggle");
+    featureCkbox.setAttribute("checked", "true");
+    const ckboxOverlay = featureBox.querySelector(".primaryInfo2 > span");
+    ckboxOverlay.classList.add("checked");
+    ckboxOverlay.classList.remove("unchecked");
+    featureBox.querySelector(".isActive").innerHTML = "Active";
+    featureBox.querySelector(".isActive").classList.remove("inactive");
+    featureBox.querySelector(".launchedDate").classList.remove("inactive");
+    featureBox.querySelector(".ownerName").classList.remove("inactive");
+    featureBox.querySelector(".ownerImg").classList.remove("imgInactive");
+
+  }
+}
+
+window.onload = function () {
+  const container = document.querySelector("#container");
+  featureContent.forEach((feature) => {
+    container.appendChild(createFeatureElement(feature))
+  });
+
+  let addEventToFeatureToggle = (featureToggle, event, fn) => {
+    Array.from(featureToggle).forEach((element) => {
+      element.addEventListener(event, fn);
+    });
+  };
+
+  const handleClick = async (e) => {
+    const featureId = e.srcElement.getAttribute("id");
+    let dataToBeSent;
+    if (e.srcElement.classList.contains("checked")) {
+      dataToBeSent = {
+        config: { enabled: false }
+      }
+    } else {
+      dataToBeSent = {
+        config: { enabled: true }
+      }
+    }
+
+    try {
+      const response = await fetch(`${BASE_URL}/featureFlags/` + featureId, {
+        method: "PATCH",
+        credentials: "include",
+        body: JSON.stringify(dataToBeSent),
+        headers: {
+          "Content-type": "application/json",
+        },
+      });
+      const result = await response.json();
+      alert(result.message);
+      changeTheFeatureToggle(featureId);
```

Comment:
Check the response code before changing the feature toggle.

What if there was an error on backend and the request returned non-successful status code.

Reply:
Verified the status code before changing the feature toggle.